src/node_buffer.cc

```diff
@@ -220,7 +220,14 @@ MaybeLocal<Object> New(Isolate* isolate,
   CHECK(actual <= length);
 
   if (actual < length) {
-    data = static_cast<char*>(realloc(data, actual));
+    char* new_data = static_cast<char*>(realloc(data, actual));
+    if ( new_data != nullptr )
+    {
+        data = new_data;
+    }else{
+        free(data);
+        data = nullptr;
+    }
```

Comment:
+0 on this change now.
See https://github.com/nodejs/node/pull/2872/files#r39474988 for reasoning.